worlds/yugioh06/logic.py

```diff
@@ -0,0 +1,16 @@
+from typing import Set
+
+from worlds.AutoWorld import LogicMixin
+
+
+class YuGiOh06Logic(LogicMixin):
+    def yugioh06_difficulty(self, player, amount: int):
+        return self.has_group("Core Booster", player, amount)
+
+    def yugioh06_has_individual(self, items: list[str], player: int):
+        amount = 0
+        for item in items:
+            if self.has(item, player):
+                amount += 1
+        return amount
+
```

Comment:
Not sure why LogicMixin was used here. These look like they could just be

```
def yugioh06_difficulty(state: CollectionState, player: int, amount: int)
```

and

```
def yugioh06_has_individual(state: CollectionState, items: list[str], player: int):
```

and then have the calls to those functions adjusted accordingly.
I think this is probably copied from somewhere, and that somewhere shouldn't be doing this either, probably